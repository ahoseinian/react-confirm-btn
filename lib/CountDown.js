// @flow

import React from 'react';

const s = {
  width: '96%',
  height: '5px',
  backgroundColor: 'red',
  position: 'absolute',
  bottom: '2px',
  left: '2%',
};

type props = {
  time: number,
};

class CountDown extends React.Component {
  props: props;
  el: HTMLElement;

  componentDidMount() {
    setTimeout(() => {
      this.el.style.width = '0';
    }, 100);
  }

  render() {
    const { time } = this.props;

    return (
      <div
        style={{ ...s, transitionDuration: time + 'ms' }}
        ref={r => {
          this.el = r;
        }}
      />
    );
  }
}

export default CountDown;
