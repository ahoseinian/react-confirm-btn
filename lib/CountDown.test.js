// @flow

import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import CountDown from './CountDown';

const wait = (ms = 100) => new Promise(r => setTimeout(() => r(true), ms));

it('renders', () => {
  const wrapper = shallow(<CountDown time={2000} />);

  expect(toJson(wrapper)).toMatchSnapshot();
});
