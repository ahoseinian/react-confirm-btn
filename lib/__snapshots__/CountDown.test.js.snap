// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  style={
    Object {
      "backgroundColor": "red",
      "bottom": "2px",
      "height": "5px",
      "left": "2%",
      "position": "absolute",
      "transitionDuration": "2000ms",
      "width": "96%",
    }
  }
/>
`;
